nemo/collections/nlp/models/nlp_model.py

```diff
@@ -312,53 +252,225 @@ def setup(self, stage: str) -> None:
                             app_state.checkpoint_callback_params,
                         )
 
-                if isinstance(self.bert_model, MegatronBertEncoder):
-                    self.bert_model.complete_lazy_init()
-
-                    # model parallel checkpoints need to be restored after torch.distributed is initialized
-                    if self._trainer.resume_from_checkpoint is not None:
-                        # update path based on model parallel rank
-                        filepath = self._trainer.resume_from_checkpoint
-                        dirname = os.path.dirname(os.path.dirname(filepath))
-                        basename = os.path.basename(filepath)
-                        filepath = f'{dirname}/mp_rank_{app_state.model_parallel_rank:02d}/{basename}'
-                        self._trainer.resume_from_checkpoint = filepath
-                        logging.info(f'Resuming training from checkpoint {self._trainer.resume_from_checkpoint}')
-                        # need to set checkpoint version for megatron-lm
-                        checkpoint_version = torch.load(self._trainer.resume_from_checkpoint).get(
-                            'checkpoint_version', None
-                        )
-                        if checkpoint_version is not None:
-                            set_checkpoint_version(checkpoint_version)
-                        else:
-                            logging.warning(
-                                'Megatron-lm checkpoint version not found. Setting checkpoint_version to 0.'
-                            )
-                            set_checkpoint_version(0)
-                    else:
-                        logging.info(
-                            f"Restoring from pretrained model parallel checkpoint: {self.bert_model._restore_path}"
-                        )
-                        self.bert_model.restore_weights(self.bert_model._restore_path)
-
-                    logging.info("Replacing sampler with model parallel sampler")
-                    mp_sampler = torch.utils.data.distributed.DistributedSampler(
-                        self._train_dl.dataset,
-                        num_replicas=app_state.data_parallel_size,
-                        rank=app_state.data_parallel_rank,
-                    )
-                    mp_dl = self._trainer.replace_sampler(self._train_dl, mp_sampler)
-                    self._train_dl = mp_dl
-                else:
-                    raise NotImplementedError(
-                        f'The BERT encoder: {self.bert_model} does not support model parallelism yet.'
-                    )
-
     def on_save_checkpoint(self, checkpoint: Dict[str, Any]) -> None:
         if hasattr(self, "bert_model") and isinstance(self.bert_model, MegatronBertEncoder):
             checkpoint['checkpoint_version'] = get_checkpoint_version()
         return None
 
+    def on_load_checkpoint(self, checkpoint: Dict[str, Any]) -> None:
```

Comment:
Is this a PTL override? Can you add a comment as to its origin - internal to nemo or overriden from nemo/ptl.

Reply:
Good catch. Comment added.